sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -427,8 +430,9 @@ case class MapObjects private(
       case _ => ""
     }
 
+    val inputDT = inputDataType.getOrElse(inputData.dataType)
```

Comment:
@cloud-fan I think there is no way to easily catch python udt here. The approach I use now is to pass a datatype (python udt's sqlType) to MapObjects.